### ISSUE TITLE
Test execution order is messed up since Spek 2.0.13

### ISSUE THREAD

**Issue by jcornaz** [JetBrains/spek#924]
Hi, 

Please consider the following code: 

```kotlin
object Spec1 : Spek({
  beforeGroup { println("before group 1") }
  beforeEachTest { println("before test 1") }
  test("test 1") { println("test 1") }
  afterGroup { println("after group 1") }
})

object Spec2 : Spek({
  beforeGroup { println("before group 2") }
  beforeEachTest { println("before test 2") }
  test("test 2") { println("test 2") }
  afterGroup { println("after group 2") }
})
```

## Using Spek `2.0.12`

I get expected standard output (in the test report created by gradle): 

For `Spec1`:
```
before group 1
before test 1
test 1
after group 1
```

For `Spec2`:
```
before group 2
before test 2
test 2
after group 2
```

## Using spek `2.0.13`

I get non-expected standard outputs:

For `Spec1`:
```
before group 1
after group 1
```

For `Spec2`:
```
before group 2
before test 1
test 1
before test 2
test 2
after group 2
```

![image](https://user-images.githubusercontent.com/2535178/93448576-0b9fd200-f8d4-11ea-8a86-639871f2910e.png)

### Personal note

I have a feeling that the root-cause is probably the same as the cause of #923.  Somehow due to execution order being changed, some tests using mock, started to fail when the full test suite is executed (but still pass when executed alone).

**Comment by raniejade** [comment]
Yeah, this was a bit unfortunate - I'm using coroutines under the hood, which explains the execution order. At any given point, a fixture or the actual test can be suspended which gives other tests to run. Lesson here is that, coroutines execution model is not really good for running tests :sweat_smile:, specially with frameworks relying on execution order. I'll rework this for `2.0.14`, making sure that execution order is at least deterministic.

**Comment by raniejade** [comment]
Should be resolved in #925
